test/integration/client/cookie-tls-proxy.js

```diff
@@ -0,0 +1,90 @@
+const Happn = require('../../..'),
+  delay = require('await-delay'),
+  HappnClient = Happn.client,
+  ChildProcess = require('child_process'),
+  path = require('path'),
+  testHelper = require('../../__fixtures/utils/test_helper').create(),
+  request = require('request');
+
+describe(testHelper.testName(__filename, 3), function() {
+  this.timeout(120e3);
+
+  it('tests the secure cookie can be grabbed based on the proxies forward headers', async () => {
+    const service = await startHappnService();
+    const proxy = await startProxyService();
+    const client = await connectClient({
+      protocol: 'https',
+      port: 11235,
+      username: '_ADMIN',
+      password: 'happn'
+    });
+    await testClient(client.session.token);
+    await client.disconnect();
+    proxy.kill();
+    service.kill();
+  });
+
+  it('tests the secure cookie can be grabbed if we are going directly to an https instance of happn', async () => {
+    const service = await startHappnService('https');
+    const client = await connectClient({
+      protocol: 'https',
+      username: '_ADMIN',
+      password: 'happn'
+    });
+    await testClient(client.session.token, 55000);
+    await client.disconnect();
+    service.kill();
+  });
+
+  function doPost(path, token, port, callback) {
+    var options = {
+      url: `https://127.0.0.1:${port || 11235}` + path,
+      method: 'POST',
+      data: { post: 'data' }
+    };
+
+    options.headers = {
+      Cookie: ['happn_token_https=' + token]
+    };
+
+    request(options, callback);
+  }
+
+  async function startHappnService(protocol) {
+    let servicePath = path.resolve(__dirname, '../../__fixtures/utils/happn-server.js');
+    let serviceProc = ChildProcess.fork(servicePath, ['-s', 'true', '-h', protocol]);
+    await delay(2e3);
+    return serviceProc;
+  }
+
+  async function startProxyService() {
+    let proxyPath = path.resolve(__dirname, '../../__fixtures/node-proxy/proxy.js');
+    let proxyProc = ChildProcess.fork(proxyPath, ['-s', 'true']);
+    await delay(2e3);
+    return proxyProc;
+  }
+
+  function testClient(token, port) {
+    return new Promise((resolve, reject) => {
+      doPost('/test/web/route', token, port, function(error, response) {
+        try {
+          testHelper.expect(error).to.eql(null);
+          testHelper.expect(response.statusCode).to.equal(200);
+          resolve();
+        } catch (e) {
+          reject(e);
+        }
+      });
+    });
+  }
+
+  function connectClient(opts) {
+    process.env['NODE_TLS_REJECT_UNAUTHORIZED'] = 0;
```

Comment:
This is going to be ignored in future releases of node - do we want to do register a certificate and do some kind and do some magic to make this ok?

Reply:
we can create a self signed cert and make it trusted by the client by giving it the ca cert - let me know if you need some help on this.